aws/resource_aws_rds_cluster_parameter_group.go

```diff
@@ -221,16 +221,33 @@ func resourceAwsRDSClusterParameterGroupUpdate(d *schema.ResourceData, meta inte
 			}
 		}
 
+		toRemove := map[string]*rds.Parameter{}
+
+		for _, p := range expandParameters(os.List()) {
+			if p.ParameterName != nil {
+				toRemove[*p.ParameterName] = p
+			}
+		}
+
+		for _, p := range expandParameters(ns.List()) {
+			if p.ParameterName != nil {
+				delete(toRemove, *p.ParameterName)
+			}
+		}
+
 		// Reset parameters that have been removed
-		parameters = expandParameters(os.Difference(ns).List())
-		if len(parameters) > 0 {
-			for parameters != nil {
+		var resetParameters []*rds.Parameter
+		for _, v := range toRemove {
+			resetParameters = append(resetParameters, v)
+		}
+		if len(resetParameters) > 0 {
```

Comment:
The changes from here on down in this file just rename the `parameters` variable to `resetParameters` in order to avoid conflicting with the previously defined `parameters` variable and make this code look more like the code in the `resource_aws_db_parameter_group.go` file.